corehq/apps/app_manager/templatetags/app_manager_extras.py

```diff
@@ -16,3 +16,9 @@ def get_available_modules_for_case_list_configuration(app, module):
             and m.module_type not in disallowed_module_types
             and m.case_type == module.case_type)
     ]
+
+
+@register.filter
+def get_available_modules_for_case_tile_configuration(app, module):
```

Comment:
i read this as [for] app [for] module. 
The logic is not just about getting all modules except this one.  There is a method above that already does that. This method gives the module that are valid options for the configuration.